scripts/update_authors.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+#  python-holidays
+#  ---------------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from pathlib import Path
+
+from git import Repo
+
+HEADER_TEMPLATE = """Python Holidays Authors
+=======================
+
+{authors}
+"""
+OUTPUT_FILENAME = "AUTHORS"
+PROJECT_DIR = Path(__file__).parent.parent
+REPOSITORY_NAME = "vacanza/python-holidays"
+
+
+class AuthorsGenerator:
+    """
+    Generates authors based on the commits metadata.
+
+    Usage example: scripts/update_authors.py
+    """
+
+    @staticmethod
+    def run():
+        """Update authors list."""
+        authors = set()
```

Comment:
```suggestion
        with open(PROJECT_DIR / OUTPUT_FILENAME, "r", encoding="UTF-8") as input_file:
            lines = input_file.readlines()
        authors = set(name.strip() for name in lines[3:])
```
If we plan to run this script from CI, with partially fetched repo.